contracts/StablecoinConverter.sol

```diff
@@ -209,20 +215,43 @@ contract StablecoinConverter is EpochTokenLocker {
                 executedSellAmount
             );
         }
-        checkAndOverrideObjectiveValue(uint(tokenConservation[0]));
+        int objectiveValue = utility - disregardedUtility;
+        checkAndOverrideObjectiveValue(objectiveValue);
         grantRewardToSolutionSubmitter(uint(tokenConservation[0]) / 2);
         checkTokenConservation(tokenConservation);
         documentTrades(batchIndex, owners, orderIds, volumes, tokenIdsForPrice);
     }
 
-    function getCurrentObjectiveValue() public view returns(uint) {
+    function getCurrentObjectiveValue() public view returns(int) {
         if (previousSolution.batchId == getCurrentBatchId() - 1) {
             return previousSolution.objectiveValue;
         } else {
-            return 0;
+            return MIN_INT;
         }
     }
 
+    function evaluateUtility(uint128 execBuy, uint128 execSell, Order memory order) internal view returns(int) {
+        // Utility = ((execBuyAmt * order.sellAmt - execSellAmt * order.buyAmt) * price.buyToken) / order.sellAmt
+        (uint128 buyAmount, uint128 sellAmount) = getBuyAndSellAmounts(order);
+        return (execBuy - ((execSell * buyAmount) / sellAmount)) * currentPrices[order.buyToken];
```

Comment:
Why do you favor this solution over this one_

        return ((execBuy * sellAmount - execSell * buyAmount) / sellAmount)) * currentPrices[order.buyToken];

I don't know which one is better... Just asking stupid questions here :)

And please use safe math

Reply:
This alternative solution does not work because of the following example:
suppose `(a, b, c, d) = (1, 2, 3, 3)`
```
(a * b * c) / d = 6 / 3 = 2
BUT
((a * b) / d) * c = 0 * 3 = 0
```